### PR TITLE
SaveGSS shouldn't segfault with unwrittable path

### PR DESCRIPTION
**Pull request by peterfpeterson**
Change so SaveGSS errors out instead. 

Description of work.

**To test:**

There is an additional test and a fix so a code review should be sufficient.

*There is no associated issue.*

*Does not need to be in the release notes* because it is an edge case that was discovered by the dev team.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/test/SaveGSSTest.h [hunk 1]
```diff
@@ -115,6 +115,28 @@ class SaveGSSTest : public CxxTest::TestSuite {
     AnalysisDataService::Instance().remove(wsName);
   }
 
+  void test_unwritablePath() {
+    if (Poco::File("/usr").exists()) {
```
Comment (ElliotAOram) [inline Framework/DataHandling/test/SaveGSSTest.h]: Unless I'm mistaken, this means this test will always be skipped on windows. Is this behaviour deliberate?

Comment (peterfpeterson) [reply Framework/DataHandling/test/SaveGSSTest.h]: You are correct. I wasn't sure of a location on windows that both exists and isn't user writable. If you have a location, I can modify the test. I don't know if it is a big deal since it is run everywhere else.

Comment (ElliotAOram) [reply Framework/DataHandling/test/SaveGSSTest.h]: Good point I can't say I know of an equivalent on windows myself either. I'm happy to approve this now I understand the reasoning.

**Review by ElliotAOram — APPROVED** [review]
